hawkular/client.py

```diff
@@ -189,7 +189,10 @@ def _http(self, url, method, data=None, decoder=None):
             res = urlopen(req, context=self.context)
 
             if res.getcode() == 200:
-                data = json.load(reader(res), cls=decoder)
+                try:
+                    data = json.load(reader(res), cls=decoder)
+                except ValueError:
+                    data = reader(res).read()
```

Comment:
This means we'd sometimes return parsed data, sometimes unparsed string, right?  (And still can give exception for malformed utf-8)
Such interfaces tend to be risky, though in this context seems mostly harmless.

- If server was supposed to return json, and returns empty / malformed, instead of clear json parse exception you get a string and fail with e.g. "TypeError: string indices must be integers, not str" downstream...
- You can't distinguish between returned json-encoded strings like `"foo\"bar"` and bare strings like `foo"bar`.

If I were writing everything from scratch, I'd move json parsing responsibility to callers.  Could have convenient interface like in Requests: where caller does `get(...).json()` it's obvious that we error if it's malformed/missing; without `.json()` (including not looking at the response at all) it's obvious we don't error (except when status is 4xx or 5xx).
@burmanm are you interested in moving to such direction long-term?

@dkorn, the only scenario you encountered is APIs that return empty body right?

1. How about not catching json parsing errors, only special-casing empty body?
   (And return empty string, None, or maybe create some special `EmptyBody` sentinel)
2. The other option I suggested to @dkorn was an optional flag whether to parse json.
   But then you have to pass it through all `_put()` etc methods.

The drawback of automagic no-errors interfaces, even (1), is one you start relying on them it's not clear from the code where you expect a body and where not.
(2) keeps it explicit, can later refactor to other styles, but from (1) it's harder to go back.